04-onion-routing.md

```diff
@@ -538,22 +535,20 @@ Next, the processing node uses the shared secret to compute a _mu_-key, which it
 in turn uses to compute the HMAC of the `hops_data`. The resulting HMAC is then
 compared against the packet's HMAC.
 
-Comparison of the computed HMAC and the packet's HMAC MUST be
-time-constant to avoid information leaks.
+Comparison of the computed HMAC and the packet's HMAC MUST be time-constant to avoid information leaks.
 
 At this point, the processing node can generate a _rho_-key and a _gamma_-key.
 
 The routing information is then deobfuscated, and the information about the
 next hop is extracted.
-To do so, the processing node copies the `hops_data` field, appends 65 `0x00`-bytes,
-generates 1365 pseudo-random bytes (using the _rho_-key), and applies the result
+To do so, the processing node copies the `hops_data` field, appends `20*FRAME_SIZE` `0x00`-bytes,
```

Comment:
This appears to be a pretty significant change. I might be missing something here, but how can how old nodes can properly generate the filler required to process these packets with this proposal?

Reply:
Ahh, actually it's fine as it's just extra random bytes append from the same stream.

Reply:
Yep, the change is because we _may_ need the extra filler once we determine the number of frames to be read. In case of old nodes it'll just use a single frame, hence the single extra frame hanging off the end is sufficient :-)